PYME/Acquire/SpoolController.py

```diff
@@ -346,7 +346,9 @@ def StartSpooling(self, fn=None, stack=False, compLevel = 2, zDwellTime = None,
         self.spooler.StartSpool()
         
         self.onSpoolStart.send(self)
-        
+
+        if self.autostart_analysis:
```

Comment:
I still think that this is the wrong place to launch analysis and that we should have the flexibility of deciding where to do it in the protocol. Doing it here will flat out not work for some use cases, and increases the risk of race conditions in others. In the cluster spooling case we need to wait until the metadata has hit the disk on the cluster and the caches have invalidated before launching analysis - this is likely to be at least 2s after we start the acquisition (1s for the timeout on the metadata put call, 1s for cache invalidation).

I do, however, see the attraction of having it toggle-able in the UI rather than having to have different protocols.

My proposed solution is that the protocols still call a `LaunchAnalysis()` method on the spool-controller (potentially renamed as, e.g. `mark_ready_for_analysis()`,  `trigger_analysis()` or similar)  when they are ready, but that the the spool controller decides what do when that happens, depending on a couple of config settings - e.g. by referencing `automatically_start_localizing` and `automatically_open_series` properties (with associated GUI controls).

I suspect you'll still object, but I don't think that there is a clean solution to this, and am pretty certain that flexibility with analysis timing is needed, particularly if we want to be flexible for other imaging modalities and types of analysis moving forward. To round things out, I'd probably call `mark_ready_for_analysis()` automatically at the end of a protocol if it hasn't already been called - i.e if you have checked 'automatically start localisation' an acquisition will launch analysis after spooling is complete, unless you flag it as being ready earlier by calling `mark_ready_for_analysis()` earlier.

Reply:
It's also worth noting that your changes quite dramatically change the way cluster spooling and analysis on the HT system work (by launching analysis at the beginning of a series rather than the end) and as such risk significant breakage. This is something I'd want to test thoroughly before use on real data and if your focus once getting back in the lab is on quickly getting the required data to finish the manuscript, this might be an un-necessary risk.

Reply:
If the protocol needs to be the authority on when analysis should be run, maybe the protocol can also provide the analysis metadata, say by adding an `mdh` kwarg to `spoolController.LaunchAnalysis`?

Needing to wait 1 s after spooling metadata doesn't break 'this', with 'this' in quotes because PR #258 refactors the localization launching to have a single entry point and a simple check there on the metadata (or 1 s, 2 s sleep) wouldn't be a big deal at all.

I also thought we'd agreed we could ditch the `automatically_open_series` concept (unless you're using open to mean something else here).

Reply:
re: changes for high-throughput system/taking data - I can easily hack in my analysis metadata to my protocol and call the spool controller method.